src/endpoint_resolver.rs

```diff
@@ -58,33 +69,32 @@ impl MomentoEndpointsResolver {
         MomentoEndpointsResolver::wrapped_endpoint("https://", hostname, ":443")
     }
 
-    fn get_control_endpoint(claims: &Claims, hosted_zone: &Option<String>) -> MomentoEndpoint {
-        MomentoEndpointsResolver::get_control_endpoint_from_hosted_zone(hosted_zone)
-            .unwrap_or_else(|| MomentoEndpointsResolver::https_endpoint(claims.cp.to_owned()))
+    fn get_control_endpoint(claims: &Claims, hosted_zone: Option<String>) -> MomentoEndpoint {
+        MomentoEndpointsResolver::get_control_endpoint_from_hosted_zone(hosted_zone).unwrap_or_else(
+            || MomentoEndpointsResolver::https_endpoint(claims.cp.as_ref().unwrap().to_owned()),
```

Comment:
so if a user does not pass in a `--endpoint` and they're using a new token they'll get a PANIC here, right? This should be cleaned up in a follow-up pr, feel free to open an issue if you want to defer

Reply:
This method will fail in isolation but in the overall scheme of things if endpoint isn't provided with new token, customers will see `"Could not parse token. Please ensure a valid token was entered correctly."`

When endpoint isn't provided, the JWT decode expects `cp` and `c` claims to be present in the token.

https://github.com/momentohq/client-sdk-rust/blob/main/src/jwt.rs#L25-L28

Reply:
Nonetheless, this should be cleaned up.
https://github.com/momentohq/client-sdk-rust/issues/61